packages/cfd/src/Containers/__tests__/investor-password-manager.spec.js

```diff
@@ -0,0 +1,135 @@
+
+const mock_errors = {
+    password: () => localize('Password should have lower and uppercase English letters with numbers.'),
+    repeated_chars_are_easy: () => localize('Repeats like "aaa" are easy to guess'),
+    repeated_patterns_are_easy: () => localize('Repeats like "abcabcabc" are only slightly harder to guess than "abc"'),
+    recent_years_are_easy: () => localize('Recent years are easy to guess'),
+};
+
+jest.mock('@deriv/shared/src/utils/validation/declarative-validation-rules.js', () => ({
+    getErrorMessages: jest.fn(() => ({
+        password_warnings: mock_errors,
+    })),
+    validLength: jest.fn(() => {
+        validLengthMock;
+    }),
+}));
+
+describe('<InvestorPasswordManager> ', () => {
+    const mock_props = {
+        error_message_investor: 'Forgot your password? Please reset your password.',
+        is_submit_success_investor: false,
+        multi_step_ref: { current: { nextStep: jest.fn() } },
+        onSubmit: jest.fn(),
+        setPasswordType: jest.fn(() => 'investor'),
+        toggleModal: jest.fn(),
+        validatePassword: jest.fn(),
+    };
+
+    it('should render the correct texts ', async () => {
+        render(<InvestorPasswordManager {...mock_props} />);
+        expect(await screen.findByText(/new investor password/i)).toBeInTheDocument();
+        expect(
+            screen.getByText(
+                /use this password to grant viewing access to another user. While they may view your trading account, they will not be able to trade or take any other actions/i
+            )
+        ).toBeInTheDocument();
+        expect(
+            screen.getByText(
+                /if this is the first time you try to create a password, or you have forgotten your password, please reset it/i
+            )
+        ).toBeInTheDocument();
+        expect(screen.getByText(/current investor password/i)).toBeInTheDocument();
+        expect(screen.getByText(/new investor password/i)).toBeInTheDocument();
+    });
+
+    it('should fill the password field and trigger the appropriate error message for repeated password pattern', async () => {
+        render(<InvestorPasswordManager {...mock_props} />);
+        expect(await screen.findByText(/new investor password/i)).toBeInTheDocument();
+        const new_investor = screen.getByLabelText(/new investor password/i);
+        await waitFor(() => {
+            fireEvent.change(new_investor, { target: { value: 'abcabcabc' } });
+        });
+        expect(
+            await screen.findByText(/repeats like "abcabcabc" are only slightly harder to guess than "abc"/i)
+        ).toBeInTheDocument();
+    });
+
+    it('should fill the password field and trigger the appropriate error message for repeated characters', async () => {
+        render(<InvestorPasswordManager {...mock_props} />);
+        expect(await screen.findByText(/new investor password/i)).toBeInTheDocument();
```

Comment:
this assertion is repeated many times in each test case, I believe it's better to test it once and remove it from all other test cases.
```suggestion
         it('should render investor password manager', async () => {
             render(<InvestorPasswordManager {...mock_props} />);
             expect(await screen.findByText(/new investor password/i)).toBeInTheDocument();
    });
```

Reply:
I need to have those in because if i dont put it, there will be an infinite loader.